vumi/message.py

```diff
@@ -27,12 +27,17 @@ def format_vumi_date(timestamp):
 def parse_vumi_date(value):
     """Parse a timestamp string using the Vumi date format.
 
+    Timestamps without microseconds are also parsed correctly.
+
     :param str value:
         The string to parse.
     :return datetime:
         A datetime object representing the timestamp.
     """
-    return datetime.strptime(value, VUMI_DATE_FORMAT)
+    date_format = VUMI_DATE_FORMAT
+    if "." not in value:
+        date_format = date_format.split(".")[0]
```

Comment:
Checking the length is `O(1)` which makes this not a micro-optimization?

I think I'm really just balking at the idea of checking all the values in our messages for `.` which feels to me like a rather unexpected thing to have happen as a result of parsing dates (of course having fields unexpectedly turned into datetimes wasn't great to begin with). I don't find the length check any less clear personally (it's just checking a different features to distinguish the two cases) but we could always add a comment if needed.

Anyway, this will be easy enough to change later so leaving it up to you.